src/drivers/wdm/queries/MultiplePagedCode/MultiplePagedCode.ql

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+/**
+ * @name MultiplePagedCode
+ * @kind problem
+ * @platform Desktop
+ * @description The function has more than one instance of PAGED_CODE or PAGED_CODE_LOCKED.
+ * @problem.severity warning
+ * @feature.area Multiple
+ * @repro.text The following code locations potentially contain dupulicate PAGED_CODE() calls.
+ * @id cpp/portedqueries/multiple-paged-code
+ * @version 1.0
+ */
+
+import cpp
+import drivers.libraries.Page
+
+//Selects routines with two at least two PAGE_CODE invocations inside one function
+from MacroInvocation mi, MacroInvocation mi2
+where
+  mi.getEnclosingFunction() = mi2.getEnclosingFunction() and
+  mi.getMacroName() = ["PAGED_CODE", "PAGED_CODE_LOCKED"] and
+  mi2.getMacroName() = ["PAGED_CODE", "PAGED_CODE_LOCKED"] and
+  mi.getLocation().getStartLine() < mi2.getLocation().getStartLine()
+select mi2,

```

Comment:
When I run this on the Windows Driver Samples, I'm seeing a number of false positives where the two instances appear to be in completely separate functions that happen to have the same name.  Did you experience this at all?

Reply:
Hmm, that's weird. I only see one result. I will call to check.

Reply:
Abenezer and I had a call about this and it seems like it's an issue on my local WDS database on my laptop.  I'm investigating but I don't think it's an issue with his code.